src/TypeScript.UnitTests/NodeJSLocator/NodeLocatorAggregatorTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarLint.VisualStudio.TypeScript.NodeJSLocator;
+
+namespace SonarLint.VisualStudio.TypeScript.UnitTests.NodeJSLocator
+{
+    [TestClass]
+    public class NodeLocatorAggregatorTests
+    {
+        private readonly Version compatibleNodeVersion = new(12, 0);
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<NodeLocatorAggregator, INodeLocator>(null, new[]
+            {
+                MefTestHelpers.CreateExport<INodeLocatorsProvider>(Mock.Of<INodeLocatorsProvider>()),
+                MefTestHelpers.CreateExport<ILogger>(Mock.Of<ILogger>())
+            });
+        }
+
+        [TestMethod]
+        public void Locate_NoLocators_Null()
+        {
+            var testSubject = CreateTestSubject();
+            var result = testSubject.Locate();
+            
+            result.Should().BeNull();
+        }
+
+        [TestMethod]
+        [DataRow(1)]
+        [DataRow(2)]
+        public void Locate_NoFoundPath_Null(int numberOfLocators)
+        {
+            var locators = new List<Mock<INodeLocator>>();
+            
+            for (var i = 0; i < numberOfLocators; i++)
+            {
+                locators.Add(SetupNodeLocator(path: null));
+            }
+
+            var testSubject = CreateTestSubject(getNodeExeVersion: null, locators: locators.Select(x=> x.Object).ToArray());
+            var result = testSubject.Locate();
+
+            result.Should().BeNull();
+
+            foreach (var locator in locators)
+            {
+                VerifyLocatorCalled(locator);
+            }
+        }
+
+        [TestMethod]
+        [DataRow(9)]
+        [DataRow(11)]
+        public void Locate_OneLocatorHasIncompatibleVersion_LocatorSkipped(int incompatibleVersion)
```

Comment:
I found this test and the next one confusing; this one in particular, because it's testing two separate things at the same time: (a) incompatible versions are detected correctly, and (b) part of the search algorithm i.e. incompatible versions are ignored. The next test checks another part of (b).

I think it would be simpler if the two aspects where tested separately e.g.
a) a data-driven test that checks that incompatible versions are recognised
b) a data-driven test that checks that compatible versions are recognised (I couldn't see a test that checks that 10 is a valid version)
c) search algorithm - stops on first valid version.

(c) could be tested in a single test e.g. with the following list of locators:
* invalid
* invalid
* valid
* invalid
* valid